linera-sdk/src/contract/runtime.rs

```diff
@@ -172,3 +192,69 @@ where
             .expect("Failed to deserialize `Response` type from cross-application call")
     }
 }
+
+/// A helper type that uses the builder pattern to configure how a message is sent, and then
+/// sends the message once it is dropped.
+pub struct MessageSender<Message>
+where
+    Message: Serialize,
+{
+    message: OutgoingMessage<Message>,
+}
+
+impl<Message> MessageSender<Message>
+where
+    Message: Serialize,
+{
+    /// Creates a new [`MessageSender`] instance to send the `message` to the `destination`.
+    pub fn new(destination: Destination, message: Message) -> Self {
+        MessageSender {
+            message: OutgoingMessage {
+                destination,
+                authenticated: false,
+                is_tracked: false,
+                grant: Resources::default(),
+                message,
+            },
+        }
+    }
+
+    /// Marks the message to be tracked, so that the sender is notified if the message is received
+    /// or receives the message back if it is rejected.
```

Comment:
I don't think we currently send a notification if it's received.
It's only about bouncing if rejected.

Reply:
Fixed, thanks!